### ISSUE TITLE
TODO: shellcheck reports error on assignment to associative array using eval

### ISSUE THREAD

**Issue by aghsmith** [koalaman/shellcheck#1387]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC1036, SC1088, SC1098
- My shellcheck version (`shellcheck --version` or 'online'): online
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

Tried on shellcheck.net and on version 0.4.7 on my own computer.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
myString="[a]=\"12\""
eval declare -A assocArray=($myString)
echo ${assocArray[a]}

```

#### Here's what shellcheck currently says:
```
Line 2	SC2034: myString appears unused. Verify use (or export if used externally).
Line 3	SC1036: '(' is invalid here. Did you forget to escape it?
Line 3	SC1088: Parsing stopped here. Invalid use of parentheses?
Line 3	SC1098: Quote/escape special characters when using eval, e.g. eval "a=(b)".
```
#### Here's what I wanted or expected to see:

TODO:
The code functions as expected if executed without shellcheck or in bash -xv
Expected output is:
12

**Comment by koalaman** [comment]
You're right. This is a quirky bash specific behavior, where words that look like assignments are parsed as such and therefore allow parentheses where they normally wouldn't be valid. It's a bit iffy to recreate, so for now I've just updated [the wiki](https://github.com/koalaman/shellcheck/wiki/SC1036).

tl;dr: Use the equivalent command `eval "declare -A assocArray=($myString)"`
